### PR TITLE
Saucelabs integration

### PR DESCRIPTION
**Pull request by alexose**
This pull request integrates Travis + Nightwatch + Saucelab and cleans up some of webpack's configuration options.

There's a bit more to do (generating the production assets could be a little better.  Also I'm not 100% sure if this breaks people's dev environments or not), but this is most of it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,44 @@
 sudo: required
-
 language: node_js
 node_js:
-  - "4.4.4"
+- 4.4.4
```
Comment (awong-dev) [inline .travis.yml]: why did we reindent the yml?  makes diffing a little annoying.

File: .travis.yml [hunk 2]
```diff
@@ -1,37 +1,44 @@
...
-      - ubuntu-toolchain-r-test
+    - llvm-toolchain-precise-3.6
+    - ubuntu-toolchain-r-test
     packages:
-      - clang-3.6
-      - g++-4.8
+    - clang-3.6
+    - g++-4.8
+  sauce_connect:
+    username: alexose
```
Comment (awong-dev) [inline .travis.yml]: do we have a dsva sauce labs account?

File: scripts/e2e-tests.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+# For running integration tests in travis.  This should happen any time we merge to staging.
+npm run webpack-prod;
```
Comment (awong-dev) [inline scripts/e2e-tests.sh]: remove extra semicolons

File: scripts/e2e-tests.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+npm run serve &
+sleep 3;
```
Comment (awong-dev) [inline scripts/e2e-tests.sh]: eek. what about writing a node.js script that loads the server.js as a module and watches for a start signal or something?

As is, this is just asking for flaky runs on travis over time.

**Comment by awong-dev** [comment]
Nifty. My main concern is with the sleep 3.  Would like to get that into something w/o the race condition before submission if it's not too hard.

File: scripts/e2e-tests.sh (hunk 4 above)
Comment (robbiethegeek) [inline scripts/e2e-tests.sh]: Yeah I would have asked the reason for  the sleep..echoing Albert here :)

File: .travis.yml (hunk 1 above)
Comment (alexose) [inline .travis.yml]: Blah.  That happened when I used `travis encrypt --yes`.  I'll set it back to how it was

File: .travis.yml (hunk 2 above)
Comment (alexose) [inline .travis.yml]: Still waiting on that via @plusjeff

**Comment by alexose** [comment]
GitHub appears to be comparing this branch against an older version of `master`.  I'm not sure why, but I created a new PR, and it seems to have fixed it:

https://github.com/department-of-veterans-affairs/healthcare-application/pull/92

**Pull request closed without merging**